src/composables/useUtilities.ts

```diff
@@ -242,12 +242,33 @@ export default function useUtilities() {
     return JSON.parse(JSON.stringify(val))
   }
 
+  /**
+   * For the label of a `required` input.
+   * If the field is required and the label ends with an '*', strip it out.
+   *
+   * @param label the label to strip
+   * @param required whether or not the field is required
+   * @returns the stripped label
+   */
+  const stripRequiredLabel = (label: string, required: boolean): string => {
+    if (!required) {
```

Comment:
question: do we need to just return an empty string if `label` is somehow undefined?

Reply:
This is the kind of stuff where I get confused, if I don't make a param optional, wouldn't someone trying to pass me undefined get a TS error trying to pass a var that has the possibility of being undefined? Same question I think for the need to use the `!!` - if I explicitly type `label` to be `string` they couldn't pass an empty array in the first place.

How do you know when you need to guard with a conditional that verifies a required param was passed and that it was the correct type and when you can you rely on TS to do the guarding implicitly?

Reply:
So yes, they would get a TS error at compile time (or type check) but we still want to prevent the app from erroring at runtime, just in case.

Reply:
So, is it like, if you are writing logic that is in a component that can be consumed by another app, you need to explicitly guard yourself. But if you're the top level app, you can rely on TS guarding? Or do you always have to explicitly guard? And if you always have to explicitly guard what's the point of using TS 😅

Reply:
It's more of a case-by-case type of thing.

Example: you write a function that accepts a single `object` parameter, but it's used all over the place in lots of different ways: sometimes the object is dynamic, originating from API response data or user input. In this scenario, maybe you can't always ensure that there isn't a way for an invalid value to be passed in. In that case, you'd include a check in the function to exit or return early in case of an invalid parameter in order to prevent it from throwing an error, and to avoid having to wrap in a try/catch, etc.